### PR TITLE
[SDTEST-901] Auto instrumentation docs for Test Optimization in Ruby

### PR DESCRIPTION
**Pull request by anmarchenko**
### What does this PR do? What is the motivation?

Adds instructions on how to auto instrument Ruby tests with Datadog Test Optimization:
- docs for supported CI providers added
- instructions on instrumenting Ruby tests changed to use autoinstrumentation approach
- manual instrumentation docs moved to the end of the document (they still relevant for really big repositories where auto instrumentation might cause a significant performance overhead)
- WebMock/VCR instructions are removed because they are obsolete (datadog-ci library doesn't require any configuration for WebMock since a long time)

### Merge instructions

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/tests/setup/ruby.md [hunk 1]
```diff
@@ -45,13 +45,23 @@ Supported test runners:
...
 ## Configuring reporting method
 
 To report test results to Datadog, you need to configure the `datadog-ci` gem:
 
 {{< tabs >}}
+{{% tab "CI Provider with Auto-Instrumentation Support" %}}
+{{% ci-autoinstrumentation %}}
+
+<div class="alert alert-warning">
+  <strong>Note</strong>: Auto-instrumentation is not supported for JRuby. If you use JRuby, please follow the manual instrumentation steps outlined below.
```
Comment (rtrieu) [inline content/en/tests/setup/ruby.md:60]:
```suggestion
  <strong>Note</strong>: Auto-instrumentation is not supported for JRuby. Follow the manual instrumentation steps outlined below instead.
```

File: content/en/tests/setup/ruby.md [hunk 2]
```diff
@@ -79,6 +94,127 @@ gem "datadog-ci", "~> 1.0", group: :test
...
+## Manual instrumentation (optional)
+
+<div class="alert alert-info">
+<strong>Attention</strong>: when using manual instrumentation, run your tests like you normally do: 
+don't change `RUBYOPT` env variable and don't prepend `bundle exec ddcirb exec` to your test command
+</div>
+
+Auto instrumentation adds additional performance overhead on code loading stage. It can be especially noticeable for 
+very big repositories with a lot of dependencies. If your project takes 20+ seconds to start, you will likely
+benefit from manually instrumenting your tests.
```
Comment (rtrieu) [inline content/en/tests/setup/ruby.md:216]:
```suggestion
to benefit from manually instrumenting your tests.
```

File: content/en/tests/setup/ruby.md [hunk 3]
```diff
@@ -79,6 +94,127 @@ gem "datadog-ci", "~> 1.0", group: :test
 
 ## Instrumenting your tests
```
[the remark this answers is not in the document]
Comment (anmarchenko) [reply content/en/tests/setup/ruby.md:95]: This is still auto instrumentation, just the one without CI provider integration. The real manual instrumentation doc is down below. Do you have any suggestions for a better title that means "Instrumenting your tests when your CI provider is not supported but still using auto-instrumentation with no code changes"? :sweat

File: content/en/tests/setup/ruby.md [hunk 4]
```diff
@@ -79,6 +94,127 @@ gem "datadog-ci", "~> 1.0", group: :test
...
+
+1. Set the following environment variables to configure the tracer:
+
+`DD_CIVISIBILITY_ENABLED=true` (Required)
+: Enables the Test Optimization product.
+
+`DD_ENV` (Required)
+: Environment where the tests are being run (for example: `local` when running tests on a developer workstation or `ci` when running them on a CI provider).
+
+`DD_SERVICE` (Required)
```
Comment (nikita-tkachenko-datadog) [inline content/en/tests/setup/ruby.md:107]: As we're transitioning to the repo-centric view, I think it makes sense to mark the service name as optional

Comment (anmarchenko) [reply content/en/tests/setup/ruby.md:107]: true

**Review by romainkomorndatadog — APPROVED** [review]

**Comment by anmarchenko** [comment]
@rtrieu this is ready for another pass

File: content/en/tests/setup/ruby.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rtrieu) [reply content/en/tests/setup/ruby.md:95]: @anmarchenko Oh sorry, misunderstood a bit. I will update my suggestion.

**Review by rtrieu — APPROVED** [review]

**Pull request merged**